### DISCUSSION TITLE
Can read, activate features but can't set Easytouch parameters

### DISCUSSION THREAD

**Discussion by xristianplante** [tagyoureit/nodejs-poolController#573] [RS485]
First off, this is a beautiful piece of software. Works great with Hubitat too!

From the Dashboard, I can see all my pool parameters. I love to see the roof solar temperature at a glance. I'm able to turn on/off Features. But I can't set parameters. When I try to set the Solar Heater Deltas, I get a "Message Aborted after 3 attempts".  

I'm not sure if that's relevant, but I'm tied to the RS485 from inside our home tapping between the Screen Logic Wireless adapter and ther Protocol Adapter.

Please, see data capture attached.

Thanks!

![OutboundMessageError](https://user-images.githubusercontent.com/111391529/184978785-819e8da4-8e0f-4dfb-93b2-7a42d53e3d57.png)
![RS485 Port](https://user-images.githubusercontent.com/111391529/184979717-04d06b75-c20c-4468-a9d6-8452ba73594c.png)
[packetLog(2022-08-16_10-36-50).log](https://github.com/tagyoureit/nodejs-poolController/files/9354473/packetLog.2022-08-16_10-36-50.log)
[config.log](https://github.com/tagyoureit/nodejs-poolController/files/9354485/config.log)
[poolConfig.log](https://github.com/tagyoureit/nodejs-poolController/files/9354486/poolConfig.log)
[poolState.log](https://github.com/tagyoureit/nodejs-poolController/files/9354487/poolState.log)
[consoleLog(2022-08-16_10-36-50).log](https://github.com/tagyoureit/nodejs-poolController/files/9354472/consoleLog.2022-08-16_10-36-50.log)

**Comment by tagyoureit** [comment]
Tapping into the SL adapter that way is fine.  It's what I did before my ET went out.

Uh oh!  We are trying to send a value of 262.  That's wrong.  

I might be able to get my Screenlogic to work again, but a bit tied up.  Are you able to do the same capture but instead change the solar heat deltas from screenlogic?  We will pick up the changes.  Please change the start temp delta from 4->5->6 (pause a few secs between each change).  Then change the stop temp delta from 5->4->3.  Finally, turn on and then off the nocturnal cooling.

BTW, if you follow [these directions](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) for doing the capture it should download a nice zip file for you.  If it doesn't download, the zip file can be found under the /

**Reply by johnny2678** [reply]
I reported something similar [here](https://github.com/tagyoureit/nodejs-poolController/issues/380)

In my case, I couldn't sniff out the right protocol packets because you can't set temp deltas in ScreenLogic (I don't think) and I couldn't get my wireless control panel to work.

**Reply by xristianplante** [reply]
@tagyoureit:

I re-loaded the Screenlogic Windows app on my PC. I hadn't used it in a while. But I can't seem to find anywhere to change the solar Deltas, either in the Screenlogc app or the Screenlogic Configurator. I'm not fully sure I understand your instructions, it seems.

Btw, I also have an issue where if I create a schedule in nodejs-poolController, it doesn't give me any error message, but it never appears in the iPhone or Windows Screenlogic apps or on the EasyTouch itself...

![Schedule ScreenLogic](https://user-images.githubusercontent.com/111391529/185204180-237fccc5-a715-42d0-9cf3-13092baec904.png)
![Schedule PoolController](https://user-images.githubusercontent.com/111391529/185204184-dd61a3cf-5c5e-4e4a-9718-72bd4f955a5e.png)

**Reply by johnny2678** [reply]
I was going to make a post this morning about schedule wonkiness with ET and njspc, but kids made me miss my replay capture time window (8AM when the pump starts for the day) 🤷‍♂️

Seeing the same thing as @xristianplante - schedules don't line up.  ex:

njspc schedule says pump is supposed to turn on at 8AM with solar heating set to 85  (this is the most recent edit to the schedule)
![image](https://user-images.githubusercontent.com/8219743/185429070-651d917e-4dac-429d-894e-6f817b649e83.png)

Screenlogic entry for that same schedule says heat setpoint of 83 (haven't used SL for schedules in ages):
![image](https://user-images.githubusercontent.com/8219743/185429940-722229ca-3dd6-4104-a62b-46d98c5c6b7d.png)

Pump turns on everyday according to the SL schedule, not njspc:
![image](https://user-images.githubusercontent.com/8219743/185430067-4d17a880-a091-431e-91d6-5c14914dd532.png)

Missed the packet capture at pump start but took one here of me changing the schedule to have an 8:10 start time and 86 degree solar heat setpoint that doesn't get pushed to ET.

I also remember something about easytouch only being able to store 12 schedules/egg timers but I only have 11 set.  

let me know if there's more info I can provide
[replay (32).zip](https://github.com/tagyoureit/nodejs-poolController/files/9375234/replay.32.zip)

**Reply by johnny2678** [reply]
hold up a tick!

not sure when, but looks like the OCP was updated to the 8:10 start time that I set via njspc in the replay above.  It's just the temp that didn't get updated.  I set the schedule to start with an 86 setpoint, OCP is still showing 83.

![image](https://user-images.githubusercontent.com/8219743/185485587-d2a716d5-d64d-4afd-9584-3bc5f5d7fbea.png)
![image](https://user-images.githubusercontent.com/8219743/185485811-a554bfc2-a04d-4242-9540-e075a6f5eda4.png)

**Comment by rstrouse** [comment]
Can I get you to change the heat setpoint on ScreenLogic during a capture?

EDIT: The key is to capture ScreenLogic doing this.

**Reply by xristianplante** [reply]
Here's what I did during the capture. I had PoolController and SL opened on my Windows. And the SL app on my iPhone.

1- On SL-Win: Increased the setpoint from 85F to 95F. 
2- Waited about a minute to see the change reflected in PoolController. I'd note that PoolController initially showed Heater Status = Cooling (with a blinking snowflake), before showing the new setpoint (Heater Status = Solar, with a sun icon). I was also monitoring this on SL-iPhone and the setpoint change appeared instantly.
3- On SL-Win: Decreased setpoint back to 85F. Waited about a minute for the change to reflect in PoolController before stopping the capture.

Is that what you wanted? I don't seem to see anything about the setpoint change in the packet capture (though I'm a  neophyte about their protocol).

[replay Plante 081922.zip](https://github.com/tagyoureit/nodejs-poolController/files/9387810/replay.Plante.081922.zip)

**Comment by rstrouse** [comment]
I have updated the software to read the outputs from ScreenLogic.  However, I need some more information about the schedules.  Can I get one of you to create a replay while changing the Heat Mode for a schedule in ScreenLogic.  I cannot seem to find any documentation on this in the EasyTouch manual.  And I didn't see any schedule messages that contain this information.

**Comment by johnny2678** [comment]
here you go... rushed a replay before heading out

in this replay
1) in njspc:
changed heat setpoint to 85
changed heat mode to heater
changed schedule start time to 8:00am

2) in screenlogic:
changed heat setpoint to 87
changed heat mode to off
changed schedule start time to 8:30am

3) in screenlogic:
changed heat setpoint to 84
changed heat mode to solar only
changed schedule start time to 8:15am

From what I can tell - changed the schedule start time in either njspc/screenlogic and it's reflected in the other app and the OCP.  Change heat setpoints/mode in njspc, it does not reflect in screenlogic/OCP.  Change heat setpoints/mode in screenlogic, it doesn't reflect in njspc, but the OCP will use whatever is set in screenlogic.

Thanks for taking a look.

[replay (33).zip](https://github.com/tagyoureit/nodejs-poolController/files/9398231/replay.33.zip)

**Reply by tagyoureit** [reply]
EasyTouch doesn't know about the heat modes or set points. SL takes care of that. This is an anomaly that I spent a while debugging.

**Reply by rstrouse** [reply]
That will explain why I can't seem to find it in the messages.  Not sure what we can do about that hack.

**Reply by johnny2678** [reply]
> EasyTouch doesn't know about the heat modes or set points. SL takes care of that. This is an anomaly that I spent a while debugging.  

Hmmm... sounds kinda like the SWG settings. No way to set output % from ET, only from ScreenLogic.

This will sound dumb but can't njspc just emulate the ScreenLogic packets when EasyTouch is detected?

**Reply by rstrouse** [reply]
ScreenLogic sends out no notification that it has changed anything.   That is why I was so confounded.  Pentair kinda tried to hack up some of the functionality into ScreenLogic it appears.  Maybe we have to remove the option for Touch controllers.  That function is native to IntelliCenter.

If this were only njsPC doing this then we could make it happen.  Unfortunately, it looks like ScreenLogic changes the setpoints after the schedule starts.  I guess if we wanted to expand the Touch controller for that function we would do exactly that but since SL seems to be in control there is no way of knowing what the setpoint was changed to or even which heat mode to use.

**Reply by johnny2678** [reply]
> If this were only njsPC doing this then we could make it happen. 

Some day...

**Reply by johnny2678** [reply]
wonder if this is stored and set by screenlogic on schedule start if I just need to do a packet capture in the AM when everything turns on.  Maybe then we'll catch the heat mode/temp packets?

**Comment by xristianplante** [comment]
I’m a bit confused. I’m not sure I understand what you called Screenlogic then. My system is just an Easytouch at my pool pad that I control with the iOS Screenlogic app. I have a protocol adapter and the wireless kit. I don’t have any dedicated device running Screenlogic. No touchpad or any input device inside our home.

**Reply by tagyoureit** [reply]
The protocol adapter and wireless kit are what is collectively called ScreenLogic. Pentair did some interesting things and SL adds some additional logic on top of EasyTouch (like changing heat modes for schedules).
